src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1168,36 +1214,52 @@ int fmi3_xml_handle_IntXXVariable(fmi3_xml_parser_context_t* context, const char
             type = (fmi3_xml_int_type_props_t*)declaredType;
         }
         variable->type = &type->super;
+        /* TODO handle this buffer clean-up in a better way */
+        {
+            const char* tmp; /* unused */
+            fmi3_xml_get_attr_str(context, fmi3_xml_elmID_Int64, fmi_attr_id_start, 0, &tmp);
+        }
+    }
+    else {
+        /* set start value */
 
-        hasStart = fmi3_xml_get_has_start(context, variable);
+        /* We must wait until after parsing dimensions, because we can't otherwise know
+         * if it's a 1x1 array or a scalar variable just by reading the start value. */
+        int hasStart = variable->startAttr != NULL;
         if (hasStart) {
-            fmi3_xml_variable_start_int_t * start = (fmi3_xml_variable_start_int_t*)fmi3_xml_alloc_variable_type_start(td, &type->super, sizeof(fmi3_xml_variable_start_int_t));
-            fmi3_int_union_t defVal;
+            jm_string startAttr = variable->startAttr;
+
+            fmi3_xml_variable_start_int_t* start = \
+                    (fmi3_xml_variable_start_int_t*)fmi3_xml_alloc_variable_type_start(
+                        td,
+                        variable->type,
+                        sizeof(fmi3_xml_variable_start_int_t));
```

Comment:
```suggestion
            fmi3_xml_variable_start_int_t* start;
            start = (fmi3_xml_variable_start_int_t*)fmi3_xml_alloc_variable_type_start(td, variable->type,
                        sizeof(fmi3_xml_variable_start_int_t));
```
At least that's how I'd format it.

Reply:
Good suggestion I have updated it. See 7912e34.